### PR TITLE
Hifi: Home Page

### LINKED ISSUE #38: Hi-fi: Home Page
- [ ] Community Signup

### PR DESCRIPTION
**Pull request by jean-zhang**
### **Note:**
- Sign up confirmation was purposely left out to later incorporate a similar component from the Contact Page
- Animation was added for the arrow, should be straight forward for other related animations
- Animation for bolding required some creativity with shadows because `font-weight` actually cannot be animated. Planning on refactoring into its own component for later reuse

### **High Level Code Overview:**
- `api.js`: updated query to now use `fetchLinks` and decrease the requests to just two, one for the home page and story data, one for just the signatures
- `index.js`, `HomePage.js`: refactored component logic to its own `HomePage` component and various child components
- `TextInputField`: refactored to take in input props

### CODE REVIEW

File: lib/api.js [hunk 1]
```diff
@@ -9,7 +9,16 @@ const client = Prismic.client(REF_API_URL);
 const DOCUMENT_TYPE = 'document.type';
 
 export async function getHomePage() {
-  const { data } = await client.getSingle('home_page');
+  const { data } = await client.getSingle('home_page', {
```
Comment (isabellberzsenyi) [inline lib/api.js:12]: beautiful

**Review by isabellberzsenyi — COMMENTED** [review]
Looks great!
One question: should the chosen featured preview image have some amount of opacity?
image
image

And a quick comment, I see you're mostly using px instead of ems .. personally I'm ok with that not sure how consistent we want to be with either one or if it matters

---
### Response:
I believe I asked about this in the client meeting. Camaryn prefers the control of editing it herself. Also the images that were chosen were unedited.

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
Everything looks great! Just a few comments - the newsletter sign up looks way smaller on the hi-fi. Could you please make it smaller? 
image
image

---
Response:
Done. Double check you're making comments as part of the review? Otherwise seems like I can't respond in the thread.

**Comment by jzhzhang66** [comment]
image

Something funky w/ the newsletter confirmation popup is happening when I click the submit button 

----
### Response: 

This was mentioned in the comment above, this is waiting for the modal to be created in a separate PR

File: style/typography.js [hunk 2]
```diff
@@ -49,8 +49,8 @@ const H3 = styled.h3`
 
 const P = styled.p`
   font-family: ${fonts.poppins};
-  font-weight: ${fontWeights.regular};
-  font-size: 1em;
+  font-weight: ${fontWeights.normal};
+  font-size: clamp(1em, 2vw, 1.2em);
```
Comment (jzhzhang66) [inline style/typography.js]: I think we should keep the font-size to be 1em - on my desktop, the clamp is making the font-size be 1.2em and it looks too big (and is also a different font size than the featured story description at the top).

Comment (jzhzhang66) [reply style/typography.js]: ^^ will also help in reducing the ellipses / newsletter sign up part to be smaller and more similar in size to hi-fi

**Comment by jzhzhang66** [comment]
Make the newsletter signup to be responsive for tablet (not cut off / there's always padding on the sides) 

image
image

**Comment by jzhzhang66** [comment]
For both of these add line-height: 1.8em to more closely mirror the spacing on the hi-fis
image
image

--- 
### Response:

Seems like something we should consider adding to the P component?

**Comment by jzhzhang66** [comment]
Add more top padding 
image
image

**Comment by jzhzhang66** [comment]
The ellipses on mobile should be a lot smaller (notice how close the lines are to the text) 
image
image

**Comment by jzhzhang66** [comment]
P.S. I really loved the animation with the arrow:)<3

**Comment by jzhzhang66** [comment]
Make sure the popup fits on mobile screen sizes 
image

**Comment by jzhzhang66** [comment]
Sorry this is really nitpicky but if you have time could you please round out the top of both arch shapes? Maybe the lower one could end somewhere around this line? Right now it looks more pointy than the design 
Also could you please make it smaller on desktop I think its a bit too big right now (rounding the top should help w/ this) 
image
image

**Review by jzhzhang66 — COMMENTED** [review]
The code looks really good I think just overall, everything feels too big so just double check with the designs to make sure they're sized similarly!

File: components/homePage/newsletterSignUp/NewsletterSignUp.styles.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+      var(--neg-shadow) var(--shadow) 0 ${colors.WHITE},
+      var(--shadow) var(--shadow) 0 ${colors.WHITE};
+    transition: all 0.8s;
+  }
+`;
+
+export const Overlay = styled.div`
+  display: ${(props) => (props.showOverlay ? 'block' : 'none')};
+  position: fixed;
+  background-color: ${colors.WHITE};
```
Comment (jzhzhang66) [inline components/homePage/newsletterSignUp/NewsletterSignUp.styles.js]: Could you please make the background-color for the overlay the creme color instead? It's less stark and is what they have on hi-fis i believe

**Comment by jean-zhang** [comment]
> The code looks really good I think just overall, everything feels too big so just double check with the designs to make sure they're sized similarly!

Bizarre, I used the exact sizes from the HiFis. I'll update.

**Review by isabellberzsenyi — APPROVED** [review]
beautiful

**Pull request merged**